src/Monitoring/AutoWatcher.coffee

```diff
@@ -0,0 +1,89 @@
+AutoWatcher =
+  init: ->
+    return unless Conf['Filter']
+
+    AutoWatcher.periodicScan()
+
+  periodicScan: ->
+    clearTimeout AutoWatcher.timeout
+
+    interval = 5 * $.MINUTE
+
+    now = Date.now()
+
+    unless Conf['autoWatchLastScan'] and (now - interval < Conf['autoWatchLastScan'])
+      AutoWatcher.scan()
+      $.set 'autoWatchLastScan', now
+    AutoWatcher.timeout = setTimeout AutoWatcher.periodicScan, interval
+
+  scan: ->
+    sitesAndBoards = (for own _, filters of Filter.filters
+      Object.keys(filter.boards) for filter in filters when filter.watch and filter.boards
+    ).flat(2).reduce (acc, i) ->
+      [_, k, v] = i.match(/(.*)\/(.*)/)
+      acc[k] ?= []
+      acc[k].push(v)
+      acc
+    , {}
```

Comment:
Needs $.dict() [which calls Object.create(null)] instead of {} here.
Should also deduplicate boards and not just sites, and ignore "siteID/*" as a board.